src/blockdata/script.rs

```diff
@@ -239,9 +239,12 @@ pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
 /// else as true", except that the overflow rules don't apply.
 #[inline]
 pub fn read_scriptbool(v: &[u8]) -> bool {
-    !(v.is_empty() ||
-        ((v[v.len() - 1] == 0 || v[v.len() - 1] == 0x80) &&
-         v.iter().rev().skip(1).all(|&w| w == 0)))
+    let last = match v.last() {
+        Some(last) => *last,
+        None => return false,
+    };
+
+    !((last == 0x00 || last == 0x80) && v.iter().rev().skip(1).all(|&b| b == 0))
```

Comment:
I think this would be an improvement.

Further, TIL about the [split_last](https://doc.rust-lang.org/std/primitive.slice.html#method.split_last) which apparently has been with us since Rust 1.5.0, and could let us get rid of the weird `iter().rev().skip(1)` dance.

Reply:
Oh, I knew about it yet I forgot when reviewing. Agree to use `split_last`.

Reply:
Done as a separate patch, previous acks are still valid :) Thanks.